src/libraries/System.Diagnostics.DiagnosticSource/tests/W3CPropagatorTests.cs

```diff
@@ -0,0 +1,337 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.RegularExpressions;
+using Xunit;
+
+namespace System.Diagnostics.Tests
+{
+    public class W3CPropagatorTests
+    {
+        private static DistributedContextPropagator s_w3cPropagator = DistributedContextPropagator.CreateDefaultPropagator();
+
+        [Fact]
+        public void TestW3CPropagatorBasics()
+        {
+            Assert.NotNull(s_w3cPropagator);
+            Assert.Equal(s_w3cPropagator, DistributedContextPropagator.CreateDefaultPropagator());
+            Assert.Equal(s_w3cPropagator, DistributedContextPropagator.Current);
+            Assert.Equal([ PropagatorTests.TraceParent, PropagatorTests.TraceState, PropagatorTests.Baggage ], s_w3cPropagator.Fields);
+        }
+
+        public static IEnumerable<object[]> W3CTestData()
+        {
+            // TraceState, Expected TraceState, input baggage, expected baggage string, expected baggage
+
+            // Simple case
+            yield return new object[] { "state=1", "state=1", new[] { new KeyValuePair<string, string>("b1", "v1") }, "b1 = v1", new[] { new KeyValuePair<string, string>("b1", "v1") } };
+
+            // Invalid trace state
+            yield return new object[] { "PassThroughW3CState=1", null, null, null, null }; // trace state key has to be lowercase
+
+            // Invalid trace state
+            yield return new object[] { "1start=1", null, null, null, null }; // trace state key has to start with lowercase
+
+            // Tabs is not allowed in trace state values. use only the valid entry
+            yield return new object[] { "start=1, end=\t1", "start=1", null, null, null }; // trace state key has to start with lowercase
+
+            // multiple trace states
+            yield return new object[] { "start=1, end=1", "start=1, end=1", null, null, null }; // trace state key has to start with lowercase
+
+            // trace state longer than the max limit
+            yield return new object[] { $"{new string('a', 255)}=1", null, null, null, null }; // trace state length max is 256
+
+            // trace state equal the max
+            yield return new object[] { $"{new string('a', 254)}=1", $"{new string('a', 254)}=1", null, null, null }; // trace state length max is 256
+
+            // Invalid baggage key.
+            yield return new object[] { null, null, new[]
+                                                    {
+                                                        new KeyValuePair<string, string>("b 1", "v1"),    // '=' is not allowed
+                                                        new KeyValuePair<string, string>("b=2", "v2"),    // Space is not allowed
```

Comment:
the comments are swapped on these first two items relative to what is being tested

Reply:
Thanks for the catch.